kotlin-eclipse-ui/src/org/jetbrains/kotlin/ui/editors/KotlinOpenDeclarationAction.java

```diff
@@ -77,7 +78,11 @@ public KotlinOpenDeclarationAction(@NotNull KotlinEditor editor) {
         super(editor.getSite());
         this.editor = editor;
         file = EditorUtil.getFile(editor);
-        javaProject = JavaCore.create(file.getProject());
+        javaProject = (file == null)?null:JavaCore.create(file.getProject());
+
+        if (file == null) {
+            KotlinLogger.log(IStatus.ERROR, "Failed to retrieve IFile from editor " + editor, null);
+        }
```

Comment:
I would like to suggest drop fields `file` and `javaProject` and retrieve them only when they will be needed.
So the signature of methods (where variables are using) will be:

```java
private SourceElement getTargetElement(@Nullable JetReferenceExpression expression, @NotNull IFile file, @NotNull IJavaProject javaProject)

private void gotoElement(@NotNull SourceElement element, @NotNull IJavaProject javaProject)
```

As a result there will be one check for null